lib/model/node.js

```diff
@@ -1,96 +1,87 @@
+var Lang = require('./../utilities/lang');
+var createClass = require('./../utilities/createClass');
+var GraphModelBase = require('./graphModelBase');
+var IOType = require('./../enum/ioType');
+
+var Port = require('./port');
+
+var validateArrayOfPortsOfType = function(ioType) {
+  return function(val) {
+    if (!Lang.isArray(val)) {
+      return false;
+    }
+
+    for (var i = 0; i < val.length; i++) {
+      if (!(val[i] instanceof Port)) {
+        return false;
+      }
+      if (val[i].get('type') !== ioType) {
+        return false;
+      }
+    }
+
+    return true; 
+  };
+};
+
 /**
  * node.js
  *
  * Definition for the Node class. A Node is a composable data structure that
  * contains a collection of Inputs and Outputs, a list of UI styles, and
  * arbitrary key-value metadata. Nodes can also contain other nodes.
- */
-
-/**
- * Creates a new node.
+ *
+ * @class Node
  * @constructor
- * 
- * @param id - The Node's unique identifier.
- * @param inputs - An array containing the Node's Inputs.
- * @param outputs - An array containing the Node's Outputs.
- * @param styles - An array of strings which contain the styles that will be
- *                 applied to the node in the visualization.
- * @param metadata - A standard JS object containing key-value metadata.
  */
-var Node = function(id, inputs, outputs, styles, metadata) {
-  this._id = id || 0;
-  this._inputs = inputs || [];
-  this._outputs = outputs || [];
-  this._styles = styles || [];
-  this._metadata = metadata || {};
-  this._x = 0;
-  this._y = 0;
-  this._pinned = false;
-};
+var Node = createClass({
+  extend: GraphModelBase,
 
-/**
- * @method getId - Getter method for the id property.
- */
-Node.prototype.getId = function() {
-  return this._id;
-};
+  constructor: function(config) {
 
-/**
- * @method getPosition - Getter method for returning an xy-coordinate pair representing the Node's
- *         position in 2D space.
- */
-Node.prototype.getPosition = function() {
-  return{
-    x: this._x,
-    y: this._y
-  };
-};
+    // re-fire events from the ports
```

Comment:
So that you can listen at the top level for any graph change deep in the graph structure.